src/test/java/org/apache/commons/jxpath/ri/compiler/ExtensionFunctionTest.java

```diff
@@ -46,9 +46,11 @@ public class ExtensionFunctionTest extends JXPathTestCase {
     private JXPathContext context;
     private TestBean testBean;
     private TypeConverter typeConverter;
+    private final String DEFAULT_ALLOW_LIST = "org.w3c.*,org.jdom.*,java.lang.String,java.util.*,org.apache.commons.*";
```

Comment:
**Note:** New Commons JXPath version will not be automatically compatible with the previous one, if the developers use functions in xpaths. All function calls will be disabled by default in the new version.

This needs to be emphasized in changelog or somewhere.

Reply:
@garydgregory Is there a default secure alow list that you can provide?

Reply:
@garydgregory As the current state of the PR fully implements the allow-list approach that denies loading any classes by default when nothing is configured, I suggest merging it so that a new release of JXPath that fixes the CVE can be published. This way users need to update their configurations with the classes they want to allow. In a follow-up change, we can introduce a default allow list of sane defaults based on the feedback we get from the users.